### DISCUSSION TITLE
Offline/Disconnected mode?

### DISCUSSION THREAD

**Discussion by ImpossibleWorks** [cpvalente/ontime#1365] [Ideas]
Are there any "disconnected" scenarios that have been thought of?

For example, I've been quite successful running OnTime on an AWS instance. *knock on wood* BUT....
if (when?) we loose internet connectivity, I'm wondering if the local OnTime "client" web page could detect that it lost connectivity with the server...and keep counting down, possibly with a small "DISCONNECTED" notification in the corner perhaps?

**Comment by alex-Arc** [comment]
Not an entirely bad idea
Some questions to start
- How long should it be able to freewheel ?
- What datasets needs updating ?
- What views should do this ? 
- How should reconciliation happen, when/if we reconnect ?

**Reply by ImpossibleWorks** [reply]
Here's how it is in my mind.

### Normal operations ###
- For timer views, OnMouseover and part of the menu (only), add "Connected to OnTime server" text in green and small somewhere. You could even do a very small green circle in the top-right corner of the screen.
-- You could even go as far as adding a setting "Show online status indicator on timer" on/off. Either way, it should always live in the MouseOver.
- For operator views, show a small green "connected" icon or "connected" type subtle text.

### At the moment of Disconnect...

**Editor/cuesheet views:**
- Handle it just like Bitfocus Companion does. Full screen gray overlay with the text like "Lost connection to server. Reconnecting..." This also disables clicking on anything since nothing's gonna work anyway.

**Views with a timer:**
- Persistent small red text at the edge of the screen with "Lost connection to server. Reconnecting...". Proper use of the blink tag LOL?
- Timer continues to count down; clock still continues to show the TOD; standby timer just sits there.
- In a situation where there is automation (meaning loading next cue, playing next cue, etc) have it perform said cues.
- In a situation where the timer is to stop at 0, have it do just that. If it's set to count-up, do that.

### At the moment of Reconnect...

**Editor/cuesheet views:**
- GUI is reset to normal operation. Gray overlay goes away, 
- Since the server was still counting down or doing its thing, it's the authoritative timekeeper. Update all the displayed data, based on the current status of whatever the server says. The server is always right.
- One scenario is that you have OnTime in the cloud and the client is at a hotel ballroom. The hotel looses internet access. An offsite producer could technically go into OnTime and update whatever. When the hotel and the client re-gains internet, the onsite client should update itself to whatever state the server is in. This means that the client could be counting down (offline) as noted above but when connection is re-established, it should follow whatever state the server is...even if it jumps time.

**Views with a timer that is _running_:**
- The "lost connection" text converts to a "Connected to OnTime server" in green then disappears after a moment.
- The time and other data displayed is updated to whatever the server says, even if it jumps/drastically changes.

Whatcha think?

**Reply by alex-Arc** [reply]
Nice, thank you for the detailed description, It all sounds reasonable
I will add it to the list, hope I can ping you when there is something to test
